src/zenml/cli/server.py

```diff
@@ -488,11 +488,6 @@ def deploy(
             TerraformServerDeploymentConfig,
         )
 
-        if isinstance(server.config, TerraformServerDeploymentConfig):
-            # TODO: maybe move the server ID into the ServerDeploymentConfig
-            #   class
-            metadata["server_id"] = str(server.config.server_id)
```

Comment:
we don't need this in the analytics anymore?

Reply:
After my changes, it's impossible to know the server ID in advance without connecting to it, which is another analytics event in itself. For this reason, I prefer to just not include it in the deployment events.